File: apps/auth-server/src/main/java/com/example/chat/auth/server/core/repository/RefreshTokenRepository.java
package com.example.chat.auth.server.core.repository;

import com.example.chat.auth.server.core.domain.RefreshToken;

import java.util.Optional;
import java.util.UUID;

public interface RefreshTokenRepository {
    Optional<RefreshToken> findByTokenValue(String tokenValue);

    void save(RefreshToken refreshToken);

    void deleteByTokenValue(String tokenValue);

    void deleteByPrincipalId(UUID principalId);
}

File: apps/auth-server/src/main/kotlin/com/example/chat/auth/server/core/repository/RefreshTokenRepository.kt
package com.example.chat.auth.server.core.repository

import com.example.chat.auth.server.core.domain.RefreshToken
import java.util.*

interface RefreshTokenRepository {
    fun findByTokenValue(tokenValue: String): Optional<RefreshToken>

    fun save(refreshToken: RefreshToken)

    fun deleteByTokenValue(tokenValue: String)

    fun deleteByPrincipalId(principalId: UUID)
}
